### ISSUE TITLE
error in running run_convert.sh

### ISSUE THREAD

**Issue by niuwenju** [anhnktp/yolov5_openvino#3]
Input:
```
./run_convert.sh
```

Output:
```
mo.utils.error.Error: Exception occurred during running replacer "REPLACEMENT_ID" (<class 'extensions.load.onnx.loader.ONNXLoader'>): Unexpected exception happened during extracting attributes for node Resize_215.
Original exception message: ONNX Resize operation from opset 12 is not supported.

```

## Environment
 - OS: [e.g. Ubuntu 18.04]
 - openvino_2020.4.287
 - onnx==1.7.0
 - torch==1.6.0+cu101
 - torchvision==0.7.0+cu101

**Comment by anhnktp** [comment]
@niuwenju  
You need to set opset=10 when convert .pt to .onnx model

**Comment by niuwenju** [comment]
by using opset=10, it can successfully convert .onnx to IR,but when running the demo by demo_openvino.py there is always None in  detections

**Comment by anhnktp** [comment]
Could you log the output from the openvino model ? Then check the output detections before and after NMS function.
Maybe you can upload your .pt or .onnx model so I can quick check.

**Comment by niuwenju** [comment]
[yolov5l.onnx](https://drive.google.com/drive/folders/1KkobgNM_-Xz6dZc4ZLph98Ktx192oWnl)

**Comment by anhnktp** [comment]
@niuwenju please share the access, I can't download your model

**Comment by niuwenju** [comment]
[yolov5l.onnx](https://drive.google.com/file/d/1kabqrEu7J4uwHWjjB1MbqW8Fw8i1ojeC/view?usp=sharing)

**Comment by anhnktp** [comment]
@niuwenju I converted succesfully your model to openvino. Could you give some image example or class of the model ?

**Comment by niuwenju** [comment]
its just the official model

**Comment by anhnktp** [comment]
which is the version of this official model (v1.0 or 2.0 or v3.0) ?

**Comment by niuwenju** [comment]
v3.0

**Comment by anhnktp** [comment]
@niuwenju lastime I converted YOLOv5 v3.0, I met the error because can not convert the Hardwish(). You should chose YOLOv5 v2.0 model, the accuracy is not much diferrent

**Comment by niuwenju** [comment]
and i convert it not by your script but official script

**Comment by niuwenju** [comment]
could you share your IR file?I would appreciate it very much.

**Comment by anhnktp** [comment]
@niuwenju currently, I'm using YOLOv5 openvino v2.0 for my work. You can use my script to convert successfully to OpenVINO. I test the YOLOv5 v3.0 (from you I just convert) and it seem miss detection (zidane.jpg)
tensor([[-0.20606, -1.72634,  0.21921,  1.56370,  2.96409, 27.00000]])
Processing time: 0.3519301414489746
[          0           0           1           0] 2.9640925 tie

**Comment by anhnktp** [comment]
I'll try convert new YOLOv5 3.0. Yes, please give your email and I'll share with you my person detection model YOLOv5 2.0

**Comment by niuwenju** [comment]
niuwenju1993@gmail.com
thanks a lot

**Comment by samba45** [comment]
> I'll try convert new YOLOv5 3.0. Yes, please give your email and I'll share with you my person detection model YOLOv5 2.0

Hi, i have an issue with my yolov5 2.0, my output from my ir model are like "-0.445" end other error... che you share plz the yolov5 2.0  so i can test it? Thanks.. my mail is sambataro3@gmail.com

**Comment by anhnktp** [comment]
@niuwenju @samba45 I've already share the YOLOv5-medium person detection model (fintuned with our private dataset)

**Comment by niuwenju** [comment]
@anhnktp there is also the same problem even use your IR file.

**Comment by anhnktp** [comment]
<img width="1028" alt="Screen Shot 2020-10-26 at 14 23 44" src="https://user-images.githubusercontent.com/30823943/97144376-e8035d80-1796-11eb-92fb-a212a05aec20.png">
This model worked well on my target dataset and normal person dataset. You should pay attention to the converted model. If you follow original repos, output from openvino model may is 3 feature maps and you need do some operation to output detection result from those feature maps like this issue.
In my openvino model, I mergeed three feature map and operation into graph and output from openvino model is list candidate detection. You just need pass this result to NMS function to have final prediction

**Comment by niuwenju** [comment]
```
from common import *
from openvino.inference_engine import IENetwork, IEPlugin


def build_argparser():
    parser = ArgumentParser(prog="demo_openvino.py")
    parser.add_argument(
        "--device",
        type=str,
        default="MYRIAD",
        help="Specify the target device to infer on; CPU, GPU, FPGA or MYRIAD is acceptable. Sample will look for a suitable plugin for device specified (CPU by default)",
    )
    parser.add_argument(
        "--model_xml",
        type=str,
        default="weights/yolov5m.xml",
        help="model.xml path",
    )
    parser.add_argument(
        "--image_path",
        type=str,
        default="test.jpg",
        help="image path",
    )
    parser.add_argument(
        "--img_size", type=int, default=640, help="inference size (pixels)"
    )
    parser.add_argument(
        "--conf_thres", type=float, default=0.2, help="object confidence threshold"
    )
    parser.add_argument(
        "--iou_thres", type=float, default=0.5, help="IOU threshold for NMS"
    )
    parser.add_argument("--cv", action="store_true", help="use PIL or opencv image")

    return parser

def w_bbox_iou(box1, box2, x1y1x2y2=True):
    if not x1y1x2y2:
        b1_x1, b1_x2 = box1[:, 0] - box1[:, 2] / 2, box1[:, 0] + box1[:, 2] / 2
        b1_y1, b1_y2 = box1[:, 1] - box1[:, 3] / 2, box1[:, 1] + box1[:, 3] / 2
        b2_x1, b2_x2 = box2[:, 0] - box2[:, 2] / 2, box2[:, 0] + box2[:, 2] / 2
        b2_y1, b2_y2 = box2[:, 1] - box2[:, 3] / 2, box2[:, 1] + box2[:, 3] / 2
    else:
        b1_x1, b1_y1, b1_x2, b1_y2 = box1[:, 0], box1[:, 1], box1[:, 2], box1[:, 3]
        b2_x1, b2_y1, b2_x2, b2_y2 = box2[:, 0], box2[:, 1], box2[:, 2], box2[:, 3]

    inter_rect_x1 = torch.max(b1_x1, b2_x1)
    inter_rect_y1 = torch.max(b1_y1, b2_y1)
    inter_rect_x2 = torch.min(b1_x2, b2_x2)
    inter_rect_y2 = torch.min(b1_y2, b2_y2)

    inter_area = torch.clamp(inter_rect_x2 - inter_rect_x1 + 1, min=0) * \
                 torch.clamp(inter_rect_y2 - inter_rect_y1 + 1, min=0)

    b1_area = (b1_x2 - b1_x1 + 1) * (b1_y2 - b1_y1 + 1)
    b2_area = (b2_x2 - b2_x1 + 1) * (b2_y2 - b2_y1 + 1)

    iou = inter_area / (b1_area + b2_area - inter_area + 1e-16)

    return iou

def w_non_max_suppression(prediction, num_classes, conf_thres=0.5, nms_thres=0.4):
    # box_corner = prediction.new(prediction.shape)
    box_corner = torch.FloatTensor(prediction.shape)
    box_corner[:, :, 0] = prediction[:, :, 0] - prediction[:, :, 2] / 2
    box_corner[:, :, 1] = prediction[:, :, 1] - prediction[:, :, 3] / 2
    box_corner[:, :, 2] = prediction[:, :, 0] + prediction[:, :, 2] / 2
    box_corner[:, :, 3] = prediction[:, :, 1] + prediction[:, :, 3] / 2
    prediction[:, :, :4] = box_corner[:, :, :4]

    output = [None for _ in range(len(prediction))]
    for image_i, image_pred in enumerate(prediction):
        conf_mask = (image_pred[:, 4] >= conf_thres).squeeze()
        image_pred = image_pred[conf_mask]

        if not image_pred.size(0):
            continue

        class_conf, class_pred = torch.max(image_pred[:, 5:5 + num_classes], 1, keepdim=True)

        # (x1, y1, x2, y2, obj_conf, class_conf, class_pred)
        detections = torch.cat((image_pred[:, :5], class_conf.float(), class_pred.float()), 1)

        unique_labels = detections[:, -1].cpu().unique()

        if prediction.is_cuda:
            unique_labels = unique_labels.cuda()

        for c in unique_labels:
            detections_class = detections[detections[:, -1] == c]
            _, conf_sort_index = torch.sort(detections_class[:, 4], descending=True)
            detections_class = detections_class[conf_sort_index]
            max_detections = []
            while detections_class.size(0):
                max_detections.append(detections_class[0].unsqueeze(0))
                if len(detections_class) == 1:
                    break
                ious = w_bbox_iou(max_detections[-1], detections_class[1:])
                detections_class = detections_class[1:][ious < nms_thres]
            max_detections = torch.cat(max_detections).data
            # Add max detections to outputs
            output[image_i] = max_detections if output[image_i] is None else torch.cat(
                (output[image_i], max_detections))

    return output

def main(args):

    num_classes = 80
    anchors = [
    [10, 13, 16, 30, 33, 23],
    [30, 61, 62, 45, 59, 119],
    [116, 90, 156, 198, 373, 326],
]  # 5s
    img_size_h = 640
    img_size_w = 640
    batch_size = 11
    model_bin = os.path.splitext(args.model_xml)[0] + ".bin"

    time.sleep(1)
    net = IENetwork(model=args.model_xml, weights=model_bin)
    input_blob = next(iter(net.inputs))
    plugin = IEPlugin(device=args.device)
    exec_net = plugin.load(network=net)

    start = time.time()

    if args.cv:
        image_src = cv2.imread(args.image_path).astype(np.float32)
        img_in = preprocess_cv_img(image_src, args.img_size, args.img_size)
    else:
        image_src = Image.open(args.image_path)
        img_in = preprocess_pil_img(image_src, args.img_size)

    outputs = exec_net.infer(inputs={input_blob: img_in})
    key0 = list(outputs.keys())[0]
    key1 = list(outputs.keys())[1]
    key2 = list(outputs.keys())[2]
    #print(outputs[key0].shape,outputs[key1].shape,outputs[key2].shape)
    outputs = [outputs[key0],outputs[key1],outputs[key2]]
    boxs = []
    a = torch.tensor(anchors).float().view(3, -1, 2)
    anchor_grid = a.clone().view(3, 1, -1, 1, 1, 2)
    for index, out in enumerate(outputs):
        out = torch.from_numpy(out)
        batch_size = 1  # using batch_size = out.shape[1] is wrong
        feature_w = out.shape[2]
        feature_h = out.shape[3]

        # Feature map corresponds to the original image zoom factor
        stride_w = int(img_size_w / feature_w)
        stride_h = int(img_size_h / feature_h)

        grid_x, grid_y = np.meshgrid(np.arange(feature_w), np.arange(feature_h))

        # cx, cy, w, h
        pred_boxes = torch.FloatTensor(out[..., :4].shape)
        pred_boxes[..., 0] = (
            torch.sigmoid(out[..., 0]) * 2.0 - 0.5 + grid_x
        ) * stride_w  # cx
        pred_boxes[..., 1] = (
            torch.sigmoid(out[..., 1]) * 2.0 - 0.5 + grid_y
        ) * stride_h  # cy
        pred_boxes[..., 2:4] = (torch.sigmoid(out[..., 2:4]) * 2) ** 2 * anchor_grid[
            index
        ]  # wh

        #pred_boxes[..., 2] = (torch.sigmoid(out[..., 2]) * 2) ** 2 * anchor_grid[index, ..., 0]  # w
        #pred_boxes[..., 3] = (torch.sigmoid(out[..., 3]) * 2) ** 2 * anchor_grid[index, ..., 1]  # h

        conf = torch.sigmoid(out[..., 4])
        pred_cls = torch.sigmoid(out[..., 5:])

        output = torch.cat(
            (
                pred_boxes.view(batch_size, -1, 4),
                conf.view(batch_size, -1, 1),
                pred_cls.view(batch_size, -1, num_classes),
            ),
            -1,
        )
        print(output)
        boxs.append(output)
    #print(len(boxs))
    outputx = torch.cat(boxs, 1)
    # NMS
    detections = w_non_max_suppression(outputx, num_classes, conf_thres=0.1, nms_thres=0.3)
    #detections = non_max_suppression(
    #    outputx, conf_thres=args.conf_thres, iou_thres=args.iou_thres, agnostic=False
    #)

    end = time.time()
    print(f"Processing time: {(end - start)}")
    if detections[0] is not None:
        display(
            detections[0], args.image_path, input_size=args.img_size, text_bg_alpha=0.6
        )

    del net
    del exec_net
    del plugin


if __name__ == "__main__":
    args = build_argparser().parse_args()

    sys.exit(main(args) or 0)
```
i have fix this problem by  doing some operation to output detection result , but the conf seems to have some error.

![test](https://user-images.githubusercontent.com/37408618/97162764-9c62ab00-17ba-11eb-8fa7-f19511692fea.jpg)

**Comment by anhnktp** [comment]
@niuwenju Yes, I met same problem of drop score when change to MYRIAD device. On CPU device, it worked normally. It is the problem of OpenVINO and it'll be fixed in the next version of OpenVINO (said by OpenVINO team)

**Comment by niuwenju** [comment]
OK, thanks a lot
